### PR TITLE
Update deprecations for v5

### LINKED ISSUE #3187: Update deprecations for next release
### Context & proposal

We need to update our deprecations to feature these two things:
- Remove ChatMarkdown
- Remove `Member:role`

### Acceptance criteria

To accept this task we need to complete the following items:

- [x] Update deprecations.

### PR DESCRIPTION
**Pull request by filbabic**
### 🎯 Goal

Fixes #3187 

### 🛠 Implementation details

Removed Markdown stuff, updated Member.role to be an error.

#### General
- [x] Assigned a person / code owner group (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/components/channels/ChannelOptions.kt [hunk 1]
```diff
@@ -98,7 +98,7 @@ public fun buildDefaultChannelOptionsState(
 ): List<ChannelOptionState> {
     val canLeaveChannel = !selectedChannel.isDistinct()
     val canDeleteChannel = channelMembers.firstOrNull { it.user.id == currentUser?.id }
-        ?.role
+        ?.channelRole
         ?.let { it == "admin" || it == "owner" }
```
Comment (adasiewiczr) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/components/channels/ChannelOptions.kt:102]: I think `channelRole` has different values comparing to `role`. You can check details [here](https://getstream.io/chat/docs/react/legacy_permissions/#channel-user-roles)

Comment (filbabic) [reply stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/components/channels/ChannelOptions.kt:102]: I think `admin` still works, but `owner` doesn't exist. So just the admin check should be fine I think

File: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/common/Extensions.kt [hunk 2]
```diff
@@ -110,7 +110,7 @@ val ChannelData.name: String
     get() = (extraData["name"] as? String) ?: ""
 
 val Member.isOwner: Boolean
```
Comment (filbabic) [inline stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/common/Extensions.kt:112]: I don't know if this still works properly or not or if it serves the same use case, we should probably merge to capabilities soon

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/common/extensions/Member.kt [hunk 3]
```diff
@@ -3,9 +3,15 @@ package io.getstream.chat.android.ui.common.extensions
 import io.getstream.chat.android.client.models.Member
 import io.getstream.chat.android.ui.common.extensions.internal.isCurrentUser
 
+/**
+ * @return if the member is the owner or an admin of a channel.
+ */
 public val Member.isOwnerOrAdmin: Boolean
-    get() = role == "owner" || role == "admin"
+    get() = channelRole == "channel_moderator"
```
Comment (filbabic) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/common/extensions/Member.kt:10]: same here

**Review by adasiewiczr — APPROVED** [review]
### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] ~New~ feature tested and works
- [x] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

**Pull request merged**